sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -200,18 +200,18 @@ class Analyzer(
   val postHocResolutionRules: Seq[Rule[LogicalPlan]] = Nil
 
   lazy val batches: Seq[Batch] = Seq(
+    Batch("Substitution", fixedPoint,
+      CTESubstitution,
+      WindowsSubstitution,
+      EliminateUnions,
+      new SubstituteUnresolvedOrdinals(conf)),
```

Comment:
I checked the order of analysis rules in branch-2.4. `Substitution` batch is after `Hints` too, like branch-3.0. Is there any change causing this issue?

Reply:
It's hard to find which change causing it for me since 2.4 and 3.0 have many differences that I don't know.

Reply:
I found the problem also exists in branch-2.4. See https://github.com/apache/spark/pull/29062#issuecomment-662233804